osu.Game/Beatmaps/WorkingBeatmap.cs

```diff
@@ -324,7 +324,7 @@ public Track Track
         public bool SkinLoaded => skin.IsResultAvailable;
         public ISkin Skin => skin.Value;
 
-        protected virtual ISkin GetSkin() => new DefaultSkin(null);
+        protected virtual ISkin GetSkin() => new BeatmapSkin(BeatmapInfo);
```

Comment:
I mentioned this on discord, but why not just make this `abstract`? Correct me if I'm wrong, but this new class you've made is *only used in tests*, at which point it should just be part of the `TestWorkingBeatmap` internals.

Reply:
Good point, will do.

Reply:
I've done this, and implemented it in all `WorkingBeatmap`s based on whether they throw `NotImplementedException` in other overrides or not, check if it's good.

I've also intentionally left the test scene although it doesn't have any test cases as it's needed in another PR (#12875), not worth going through the hassle of moving this from here to there.

Reply:
I've just removed this completely for now. A test scene is not the place to fix this issue (as it also affects every other usage of the `Transformer` logic).

Reply:
Good choice, I'll get a PR ready for it then, should be pretty simple to understand here.

Reply:
I've instead pushed it directly to #12875 since they both rely on each other, #12875 requires the transformers fix for its test scene to not false-pass, and the transformers fix requires #12875 because the bug will become visible and affect users if #12875 isn't merged yet.